internal/pkg/list/list.go

```diff
@@ -0,0 +1,167 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package list
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"math"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/aws/copilot-cli/internal/pkg/config"
+)
+
+const (
+	// Display settings.
+	minCellWidth           = 20  // minimum number of characters in a table's cell.
+	tabWidth               = 4   // number of characters in between columns.
+	cellPaddingWidth       = 2   // number of padding characters added by default to a cell.
+	paddingChar            = ' ' // character in between columns.
+	noAdditionalFormatting = 0
+
+	// Workload types.
+	jobWorkloadType = "job"
+	svcWorkloadType = "service"
+)
+
+// Store wraps the methods required for interacting with config stores.
+type Store interface {
+	GetApplication(appName string) (*config.Application, error)
+	ListJobs(appName string) ([]*config.Workload, error)
+	ListServices(appName string) ([]*config.Workload, error)
+}
+
+// Workspace wraps the methods required to interact with a local workspace.
+type Workspace interface {
+	JobNames() ([]string, error)
+	ServiceNames() ([]string, error)
+}
+
+// Lister handles
+type Lister struct {
+	store Store
+	ws    Workspace
+	w     io.Writer
+}
+
+// NewLister returns a new Lister structure, which can be used to list Jobs or Services
+// from both the config store and workspace.
+func NewLister(ws Workspace, store Store, writer io.Writer) *Lister {
+	return &Lister{
+		store: store,
+		ws:    ws,
+		w:     writer,
+	}
+}
+
+// Jobs lists all jobs, either locally or in the workspace, and writes the output to a writer.
+func (l *Lister) Jobs(appName string, showLocal bool, writeJSON bool) error {
```

Comment:
Boolean parameters are usually a hint that there is a chance for us to refactor the method! In this situation I think we can leverage: [Replace Conditional with Polymorphism](https://refactoring.com/catalog/replaceConditionalWithPolymorphism.html) nicely.

```go
type JobListWriter struct {
	FilterOnLocal bool
	JSONFormat bool

	Store JobsStore
	WS    JobsWorkspace
	Writer     io.Writer
}

func (w *JobListWriter) Write() error

type SvcListWriter struct {
	FilterOnLocal bool
	JSONFormat bool

	Store ServicesStore
	WS    ServicesWorkspace
	Writer     io.Writer
}

func (w *SvcListWriter) Write() error
```
This way both structs implement the same `Write() error` interface.